lib/MusicBrainz/Server/Data/Collection.pm

```diff
@@ -429,6 +429,20 @@ sub set_collaborators {
         }, @$collaborators
     );
 
+    # Remove non-owner, no-longer-collaborator subscriptions if collection is private
+    $self->sql->do(
+        'DELETE FROM editor_subscribe_collection esc
+         USING editor_collection ec
+         WHERE esc.collection = ?
+         AND esc.collection = ec.id
+         AND ec.public IS FALSE
+         AND esc.editor != ec.editor
+         AND esc.editor NOT IN (
+             SELECT editor FROM editor_collection_collaborator
+             WHERE collection = esc.collection
+        )',
```

Comment:
In `del_collection_sub_on_private` we do `UPDATE ... SET available = FALSE, last_seen_name = OLD.name`.  Is there a reason we delete them here instead?

Reply:
Honestly, probably just that I didn't know that code existed back then. I should probably do the same here then, or write a SQL function for that?

Reply:
Let's just update the query here then.  If we need it more than twice let's add a function. :)

Reply:
Done, I think it should be working. Added a test for this use case too.